### PR TITLE
[SecurityBundle] Move configuration from XML to PHP

### PR DESCRIPTION
**Pull request by JudicaelR**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | #37186 
| License       | MIT
| Doc PR        |- 

Move security configuration from XML to PHP for `collectors`, `console` and `guard`

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/Resources/config/collectors.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Bundle\SecurityBundle\DataCollector\SecurityDataCollector;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('data_collector.security', SecurityDataCollector::class)
+        ->args([
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Resources/config/collectors.php:19]: args and tag should be indented deeper than set

File: src/Symfony/Bundle/SecurityBundle/Resources/config/console.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Bundle\SecurityBundle\Command\UserPasswordEncoderCommand;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('security.command.user_password_encoder', UserPasswordEncoderCommand::class)
+        ->args([service('security.encoder_factory'), []])
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Resources/config/console.php:19]: indention is wrong

Comment (Tobion) [reply src/Symfony/Bundle/SecurityBundle/Resources/config/console.php:19]: please use `abstract_arg('encoders user classes')` for the second argument and on a new line

File: src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Component\Security\Guard\Firewall\GuardAuthenticationListener;
+use Symfony\Component\Security\Guard\GuardAuthenticatorHandler;
+use Symfony\Component\Security\Guard\Provider\GuardAuthenticationProvider;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('security.authentication.guard_handler', GuardAuthenticatorHandler::class)
+            ->call('setSessionAuthenticationStrategy', [service('security.authentication.session_strategy')])
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php:21]: please put `call` after `args`

File: src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                service('event_dispatcher')->nullOnInvalid(),
+                abstract_arg('stateless firewall keys'),
+            ])
+
+        ->alias('Symfony\Component\Security\Guard\GuardAuthenticatorHandler', 'security.authentication.guard_handler')
+
+        ->set('security.authentication.provider.guard', GuardAuthenticationProvider::class)
+            ->abstract()
+            ->args([
+                abstract_arg('Simple Authenticator'),
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php:33]: Just `Authenticators` as its iterable and there is no "simple" (yes it already seems wrong in the original xml)

File: src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                service('security.password_encoder'),
+            ])
+
+        ->set('security.authentication.listener.guard', GuardAuthenticationListener::class)
+            ->abstract()
+            ->args([
+                service('security.authentication.guard_handler'),
+                service('security.authentication.manager'),
+                abstract_arg('Provider-shared Key'),
+                abstract_arg('Authenticator'),
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php:46]: `Authenticators` as its iterable

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/SecurityBundle/Resources/config/collectors.php (hunk 1 above)
Comment (JudicaelR) [reply src/Symfony/Bundle/SecurityBundle/Resources/config/collectors.php]: Done

File: src/Symfony/Bundle/SecurityBundle/Resources/config/console.php (hunk 2 above)
Comment (JudicaelR) [reply src/Symfony/Bundle/SecurityBundle/Resources/config/console.php]: Done

File: src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php (hunk 3 above)
Comment (JudicaelR) [reply src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php]: Done

File: src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php (hunk 4 above)
Comment (JudicaelR) [reply src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php]: Done

File: src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php (hunk 5 above)
Comment (JudicaelR) [reply src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php]: Done

File: src/Symfony/Bundle/SecurityBundle/Resources/config/collectors.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                service('security.role_hierarchy'),
+                service('security.logout_url_generator'),
+                service('security.access.decision_manager'),
+                service('security.firewall.map'),
+                service('debug.security.firewall')->nullOnInvalid(),
+            ])
+            ->tag('data_collector', [
+                'template' => '@Security/Collector/security.html.twig',
+                'id' => 'security',
+                'priority' => '270',
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Resources/config/collectors.php:30]: please make 270 an int

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    $container->services()
+        ->set('security.authentication.guard_handler', GuardAuthenticatorHandler::class)
+            ->args([
+                service('security.token_storage'),
+                service('event_dispatcher')->nullOnInvalid(),
+                abstract_arg('stateless firewall keys'),
+            ])
+            ->call('setSessionAuthenticationStrategy', [service('security.authentication.session_strategy')])
+
+        ->alias('Symfony\Component\Security\Guard\GuardAuthenticatorHandler', 'security.authentication.guard_handler')
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php:28]: GuardAuthenticatorHandler::class

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php (hunk 7 above)
Comment (JudicaelR) [reply src/Symfony/Bundle/SecurityBundle/Resources/config/guard.php]: Done

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @JudicaelR.

**Pull request merged**